ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+        list of flattened list
+    Notes
+    -----
+    Format of the output:
+    [{Key1:Value1, Key2:Value2, Key3:Value3}]
+    """
+
+    res1 = list()
+    for entry in lst:
+        for key, value in entry.items():
+            res = dict()
+            res['name'] = key
+            for k,v in value.items():
+                res[k] = v
+        res1.append(res)
+    return res1
+
+
+
+def _insert_in_metadata_fits_safe(meta, key, value):
+    """Helper function to insert key-value pair into metadata in a way that
+       FITS can serialize
+
+    Parameters
+    ----------
+    key : str
+        Key to be inserted in the dictionary
+    value : str or None
+        Value to be inserted
+
+    Notes
+    -----
+    This addresses a shortcoming of the FITS standard. There are length
+        restrictions on both the ``key`` (8 characters) and ``value`` (72
+        characters) in the FITS standard. There is a convention for handling
+        long keywords and a convention for handling long values, but the
+        two conventions cannot be used at the same time.
+
+        This addresses that case by checking the length of the ``key`` and
+        ``value`` and, if necessary, shortening the key.
+    Reference
+    ---------
+    This helper method is taken from `astropy.nddata.cddata`. Not imported
+    as this is a private method, subject to change
+
+    """
+
+    if len(key) > 8 and len(value) > 72:
+        short_name = key[:8]
+        meta['HIERARCH {0}'.format(key.upper())] = (short_name, "Shortened name for {}".format(key))
+        meta['short_name'] = value
+    else:
+        meta[key] = value
+
+
+class NDCubeIOMixin(NDIOMixin):
+
+    # Inherit docstring from parent class
+    __doc__ = NDIOMixin.__doc__
+
+    MISSING = 'MISNG{0}'
+    EXTRA_COORDS = 'EXTCR{0}'
+
+    # Here we create a read and write function to support I/O of NDCube files
+    # Currently read option is not supported
+    # Here we delegate the write method to save the files to different helper functions
+    # For different formats, we can define different helper functions, and based on the file type
+    # we can call them. Currently only FITS is supported.
+
+    # Here we create a constructor for a mixin class, which is quite rare for a mixin
+    # Since the data of the NDCube needs to be used in this mixin, the
+    # constructor seems to be useful
+
+    def __init__(self, data, wcs, uncertainty, mask, meta,
+                 unit, copy, extra_coords=None, missing_axis=None, **kwargs):
+
+        super().__init__(data, wcs=wcs, uncertainty=uncertainty, mask=mask,
+                         meta=meta, unit=unit, copy=copy, **kwargs)
+        self.__data = data
+        self.__wcs = wcs
+        self.__meta = meta
+        self.__uncertainty = super().uncertainty
+        self.__uncertainty_type = None
+        self.__mask = mask
+        self.__unit = unit
+        self.__missing_axis = missing_axis
+        self.__extra_coords = extra_coords
```

Comment:
I don't think it's necessary to create all these copies of the attributes?  Just refer to the attributes themselves in the ```write``` function.  The one exception here is ```self.extra_coords``` is not the same to the input ```extra_coords```.  But ```self.extra_coords``` can be converted back into input form using ```utils.cube.convert_extra_coords_dict_to_input_format``` (https://github.com/sunpy/ndcube/blob/master/ndcube/utils/cube.py#L106)

Reply:
We cannot access these data from a function, can we? So I had to make a constructor, in order to store the instance variables. Since the instance variables of `NDCube` uses `_<variable_name>`, I intentionally used variables different from the above.

Reply:
But this a method on a mixin class, rather than a function, isn’t it? If so, all the attributes of the NDCube will be available within the method.

Reply:
Yeah, we can certainly access the instance attributes.